Training Reproduction Fixes

marwankefah opened the pull request:
Hi, I was running your code to reproduce it but I came across 

1) TypeError update in test an unexpected keyword argument "thresh", and I fixed it with necessary updates.

2) I also return precision/recall for visualization on tensorboard

3) I also fixed the path 'cares_SVHN' and fixed the directory structure for SVHN with makedirs. (should be populated to all)

4) I advice also that  time_current = str(int(time.time())) instead of time_current = 'attempt0'

May you let me know if I misunderstood something.

BR,
Marawan Elbatel

marwankefah committed:
SVHN experiment: TypeError update in test an unexpected keyword argument 'thresh, makedirs in SVHN dataset, and log folder creation

marwankefah committed:
un_dist,un_dist_onlyunsup needed for training?

xxliang99 commented:
Hi, thanks for your interest. 

The arguments un_dist and un_dist_onlyunsup are not needed in training. They refer to the choice of temporal intra-subset averaged model (theta_avg) in DMA. We tried from intra-subset averaged model, previous global model, and another mixed type, in which the first one gave the best performance, which is reported in the paper.

Therefore, the arguments are no longer needed in the final code, where only the "avg" branch is reserved.

marwankefah commented:
Thank you for your prompt response, your code is clear to understand, and I can see really lots of effort done into it. 

My concern is that they are not used logically but they are used In line with the code in L285 in train_main.py

I came across these **errors** when I tried to reproduce some/all experiment (I added Colab links to README) 

1. cares_SVHN not found (discrepancy between the SummaryWriter and the mkdir folder name) 
2. TypeError: epochVal_metrics_test() got an unexpected keyword argument 'thresh'.
3. args.un_dist and args.un_dist_onlyunsup are not found (mentioned in L285).
4. CUDA_VISIBLE_DEVICES should be mentioned before initializing torch (I abstracted it in the options)
6. Discrepancies between partition_strategy/{#$!@$}.pth and code implementation in test.py
7. Sending train_idx and test_idxs in case of skin dataset in partition_data_allnoniid.

I fixed them for reproduction and I added colabs to README to reproduce your results (train/test).

BR,
Yours Sincerely,
Marawan Elbatel

marwankefah committed:
added labeled num to the sup_size calculation

marwankefah committed:
fixed partition file path '_ordered???'

marwankefah committed:
Update README.md

marwankefah committed:
updated test logging

marwankefah committed:
fixed Skin dataset Errors and Code Architecture for reproducing

marwankefah committed:
Merge remote-tracking branch 'origin/main' into main

marwankefah committed:
abstracted some options

marwankefah committed:
Update README.md

marwankefah committed:
abstracted partition_data_allnoniid

marwankefah committed:
Merge remote-tracking branch 'origin/main' into main

marwankefah committed:
edit model output path with time_current

marwankefah committed:
updated model path
